File: Unity/Assets/Bunny.cs
using UnityEngine;


public class Bunny : MonoBehaviour
{
  float speedY;
  float speedX;

  float minX;
  float maxX;
  float minY;
  float maxY;

  float gravity = 0.008f;

  private void Start()
  {

    speedY = Random.Range(-5f, 5f) / 60;
    speedX = Random.Range(0f, 10f) / 60;

    // get bounds
    minX = 0;
    minY = 0;

    maxY = -(2f * Camera.main.orthographicSize);
    maxX = 2f * Camera.main.orthographicSize * Camera.main.aspect;
  }

  private void Update()
  {
    // Move
    transform.position += new Vector3(speedX, speedY, 0);
    speedY -= gravity;

    if (transform.position.x > maxX)
    {
      speedX *= -1;
      transform.position = new Vector2(maxX, transform.position.y);
    }
    else if (transform.position.x < minX)
    {
      speedX *= -1;
      transform.position = new Vector2(minX, transform.position.y);
    }

    if (transform.position.y < this.maxY)
    {
      speedY *= -0.85f;
      transform.position = new Vector2(transform.position.x, maxY);
      transform.eulerAngles = new Vector3(0, 0, Random.Range(-6f, 6f));
      if (Random.Range(0f, 1f) > 0.5f)
      {
        speedY += Random.Range(0f, .1f);
      }
    }
    else if (transform.position.y > this.minY)
    {
      speedY = 0;
      transform.position = new Vector2(transform.position.x, minY);
    }
  }
}

File: Phaser/src/scenes/Bunny.ts
import Phaser from "phaser";

const gravity = 0.5;
const minX = 0;
const maxX = 800;
const minY = 0;
const maxY = 600;

export default class Bunny extends Phaser.GameObjects.Sprite {
  speedX = Math.random() * 10;
  speedY = Math.random() * 10 - 5;

  constructor(scene: Phaser.Scene) {
    super(scene, 10, 10, "bunny");
    // speedX = Math.random() * 10;
    // speedY = Math.random() * 10 - 5;
    this.setOrigin(0.5, 1);
    scene.add.existing(this);
  }

  preUpdate() {
    this.x += this.speedX;
    this.y += this.speedY;
    this.speedY += gravity;

    if (this.x > maxX) {
      this.speedX *= -1;
      this.x = maxX;
    } else if (this.x < minX) {
      this.speedX *= -1;
      this.x = minX;
    }

    if (this.y > maxY) {
      this.speedY *= -0.85;
      this.y = maxY;
      this.rotation = (Math.random() - 0.5) * 0.2;
      if (Math.random() > 0.5) {
        this.speedY -= Math.random() * 6;
      }
    } else if (this.y < minY) {
      this.speedY = 0;
      this.y = minY;
    }
  }
}
